File: _REPO/APRESS/distributed-.net-programming-in-vb-.net/Chapter03/SimpleContext/SimpleContext.vb
Imports System.Runtime.Remoting 'RemotingServices class
Imports System.Runtime.Remoting.Contexts 'Synchronization attribute
	
Module SimpleContext

   Sub Main()
      Dim myBound As New MyContextBoundClass()
      Dim myAgile As New MyAgileClass()

      'Are they in or out of context?
      Console.WriteLine()
      Console.WriteLine("Is myBound out of context? {0}", _
         RemotingServices.IsObjectOutOfContext(myBound))

      Console.WriteLine("Is myAgile out of context? {0}", _
         RemotingServices.IsObjectOutOfContext(myAgile))

      'Direct reference or proxy?
      Console.WriteLine()
      Console.WriteLine("Is myBound a proxy? {0}", _
         RemotingServices.IsTransparentProxy(myBound))

      Console.WriteLine("Is myAgile a proxy? {0}", _
         RemotingServices.IsTransparentProxy(myAgile))

      Console.ReadLine()
   End Sub

End Module

'A context bound class
<Synchronization()> _
Public Class MyContextBoundClass 
   Inherits ContextBoundObject
End Class

'This is a context agile class
Public Class MyAgileClass
End Class
	   

File: _REPO/APRESS/distributed-.net-programming-in-csharp/Chapter03/SimpleContext/SimpleContext.cs

namespace SimpleContext
{
   using System;
   using System.Runtime.Remoting.Contexts; // Synchronization attribute
   using System.Runtime.Remoting;          // RemotingServices class
   
	class SimpleContextMain
	{
		static void Main(string[] args)
		{
			MyContextBoundClass myBound = new MyContextBoundClass();
         MyAgileClass myAgile = new MyAgileClass();

         // Are they in or out of context?
         Console.WriteLine("\nIs myBound out of context? {0}",
            RemotingServices.IsObjectOutOfContext(myBound));

         Console.WriteLine("Is myAgile out of context? {0}",
            RemotingServices.IsObjectOutOfContext(myAgile));

         // Direct reference or proxy?
         Console.WriteLine("\nIs myBound a proxy? {0}",
            RemotingServices.IsTransparentProxy(myBound));

         Console.WriteLine("Is myAgile a proxy? {0}",
            RemotingServices.IsTransparentProxy(myAgile));
  
         Console.ReadLine();
		}
	}

   // A context bound class
   [Synchronization]
   public class MyContextBoundClass : ContextBoundObject
   {}
   
   // This is a context agile class
   public class MyAgileClass
   {}    
}
